src/tests.py

```diff
@@ -236,13 +238,127 @@ def _integration_test(target_directory):
             assert str(e) == "Branch does not have a parent: master"
 
 
+
+def _test_clean_branches(target_directory):
+    # type: (Text) -> None
+    target_directory = os.path.expanduser(target_directory)
+    target_container = os.path.dirname(target_directory)
+    assert not os.path.exists(target_directory)
+    assert os.path.isdir(target_container)
+
+    with run_test(target_directory):
+        # Initialize a repo and add a first commit so we can tell what branch we're on.
+        print("Initializing repo")
+        git("init")
+        open("hello.txt", "w").close()
+        git("add .")
+        git("commit -am initial_commit")
```

Comment:
Similarly would be good to pull this out into a `_initialize_repo` function.

Reply:
👍